opentelemetry/src/logs/record.rs

```diff
@@ -1,6 +1,9 @@
 use crate::{Key, StringValue};
 use std::{borrow::Cow, collections::HashMap, time::SystemTime};
 
+#[cfg(feature = "trace")]
```

Comment:
i think we should remove the feature gate for these. User maybe using a different tracing system, and still should be able to correlate logs to that tracing system. Hence, the basic primitives like TraceId,SpanId etc. should be always available, not just when "trace" feature flag is enabled.

@TommyCpp @lalitb what do you think?

Reply:
Yes, I agree that we can remove the trace flag. Since TraceContext/SpanContext will be used in both the logging and metrics implementations, I don't see a scenario where it could be conditionally disabled.

Reply:
The feature is unconditionally pulled in by `opentelemetry_sdk`, but is optional in `opentelemetry`. I looked at making `TraceId`, `SpanId`, `TraceFlags` not conditional on the `trace` feature, but it meant also making other types like `TraceError` available too. That seemed like a bigger change so didn't go down that road without discussion.